lib/c-api/build.rs

```diff
@@ -89,8 +87,41 @@ fn build_wasm_c_api_headers(crate_dir: &str, out_dir: &str) {
     out_header_file.push("wasmer_wasm");
 
     let mut pre_header = format!(
-        r#"// The Wasmer C/C++ header file compatible with the `wasm-c-api` standard API.
-// This file is generated by lib/c-api/build.rs.
+        r#"// The Wasmer C/C++ header file compatible with the [`wasm-c-api`]
+// standard API, as `wasm.h` (included here).
+//
+// This file is automatically generated by `lib/c-api/build.rs` of the
+// [`wasmer-c-api`] Rust crate.
+//
+// # Stability
+//
+// The [`wasm-c-api`] standard API is a _living_ standard. There is no
+// commitment for stability yet. We (Wasmer) will try our best to keep
+// backward compatibility as much as possible. Nonetheless, some
+// necessary API aren't yet standardized, and as such, we provide a
+// custom API, e.g. `wasi_*` types and functions.
+//
+// The documentation makes it clear whether a function is unstable.
+// 
+// When a type or a function will be deprecated, it will be marked as
+// such with the appropriated compiler warning, and will be removed at
+// the next release round.
```

Comment:
We need a tool for keeping track of what we said we'd remove. We have deprecated items in our Rust APIs (for example wasmer-wasi) saying that they should be removed before 1.0.0 launch but they're still there

Reply:
We should just mark them as `#[deprecated]` I think, and remove all the types and functions before the release. Thoughts?

Reply:
Yeah, I'm saying that the function marked with that attribute wasn't removed before 1.0.0; I don't think we should add more manual, error-prone steps to the release to check all `deprecated` functions too, because in all likelihood that'll cause problems that need to be resolved and we'll just end up with a blocked release. We need like notifications a few days in advance of a release to figure those out I think. Not strictly related to this PR, but it's something we haven't figured out yet